frontend/lib/screens/medicationinfo.dart

```diff
@@ -0,0 +1,84 @@
+// lib/screens/medication_info_screen.dart
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+import 'package:intl/intl.dart';
+
+class MedicationInfoScreen extends StatefulWidget {
+  final int userID;
+  const MedicationInfoScreen({super.key, required this.userID});
+
+  @override
+  _MedicationInfoScreenState createState() => _MedicationInfoScreenState();
+}
+
+class _MedicationInfoScreenState extends State<MedicationInfoScreen> {
+  List meds = [];
+
+  @override
+  void initState() {
+    super.initState();
+    fetchMedicationInfo();
+  }
+
+  Future<void> fetchMedicationInfo() async {
+    try {
+      // print('${widget.userID}');
+      // print(
+      //    'Fetching data from: http://192.168.2.14:8000/api/medication_info/${widget.userID}/');
+      final response = await http.get(Uri.parse(
+          'http://192.168.2.14:8000/api/medication_info/${widget.userID}/'));
+      // print('Response status code: ${response.statusCode}');
+      // print('Response body: ${response.body}');
+      if (response.statusCode == 200) {
+        setState(() {
+          meds = json.decode(response.body);
+        });
+      } else {
+        throw Exception(
+            'Failed to load medication info: ${response.statusCode}');
+      }
+    } catch (e) {
+      // print('Error fetching medication info: $e');
+      // Handle error state or display an error message to the user
+    }
+  }
+
+  String convertTo12HourFormat(String time24) {
```

Comment:
I like this, if your dates get any funkier you could also find a library to handle them, but why overcomplicate things now. Great job.